### PR TITLE
exploratory NN pipeline

### PR DESCRIPTION
**Pull request by aappling-usgs**
I did some free-form model exploration for Mendota last week but then started getting frustrated with the need to remember or manually jot down RMSEs and observations about model fit for each model variant, so here's a bit of a pipeline to support exploration with the Mendota dataset. I'm also hoping that this pipeline will be a start toward modeling multiple lakes, but in the meantime I've been focusing on model diagnostics and comparison to help us choose a model structure.

@wdwatkins , I borrowed heavily from your src/test_nn.R and src/functions.R throughout this exercise. Thanks for pioneering the implementation!

The basic structure:
* 1_format assumes (not committed) the files `1_format/in/Mendota_meteo.feather`, `1_format/in/output.nc`, and `1_format/tmp/mendota_obs.csv`. I think these three files will be familiar to David if not others, and I hope they're not too different from what we'll get from pipelines 1 and 2 in the future.
* 1_format does some data processing to produce NN-model-ready data, including reading in the predictors, observations, and GLM predictions; computing features just as Anuj and David have done; formatting the data according to the intended model structure (NN, RNN, spatially explicit variants) (I've only tested NN models so far, though); splitting the data into train-develop-test partitions, and scaling all partitions according to the training data.
* 2_model contains 2_model/src/models.R, which defines a function per model variant we want to try. There's some redundancy here, but I think a lot of flexibility to quickly create new model structures that may or may not look much like models we've already tried.
* Right now you have to add a couple of remake targets each time you add a model function. Then you can run scmake on the eval target, e.g., `scmake('model_03_eval')` to build and report on model 3
* The report is defined in 2_model/src/assessment.Rmd and produces a new html output file for each new eval target you create.  This is how I can tell you with some confidence that model 2 (with L2 regularization only and Adam optimization) slightly outperforms model 1 (with L1 and L2), and that 30% dropout in the hidden layers is not working, at least how I've implemented it in model 3.

I'll share these initial reports in slack because github doesn't allow html attachments.

### CODE REVIEW

File: 2_model/src/evaluate.R [hunk 1]
```diff
@@ -0,0 +1,7 @@
+evaluate_model <- function(model_list, dat, rmd_file) {
+  rmarkdown::render(
```
Comment (wdwatkins) [inline 2_model/src/evaluate.R]: cool, haven't rendered rmarkdown programatically before

File: 2_model/src/models.R [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    loss = "mean_squared_error")
+
+  # fit the model
+  history <- network %>% fit(
+    dat$train_input, dat$train_target, epochs = 10000, batch_size = 1000,
+    callbacks = callback_early_stopping(patience = 500, monitor = "loss"),
+    validation_split = 0.1)
+
+  # package and return the results
+  return(list(id=1, description="as in Anuj's paper", serialized=serialize_model(network), history=history))
```
Comment (wdwatkins) [inline 2_model/src/models.R]: 👍
